wsapi/wsapiV2.go

```diff
@@ -736,24 +708,196 @@ func HandleV2RawData(state interfaces.IState, params interface{}) (interface{},
+			directoryBlockHeight = ecBlock.GetDatabaseHeight()
+		} else if fBlock, _ := db.FetchFBlock(hash); fBlock != nil {
+			directoryBlockHeight = fBlock.GetDatabaseHeight()
+		} else if tx, _ := state.FetchECTransactionByHash(hash); tx != nil {
+			entryHash := tx.GetEntryHash()
+			dBlockHash, err := db.FetchIncludedIn(entryHash)
+			if err != nil {
+				return nil, NewInternalDatabaseError()
+			}
+			eBlock, err := db.FetchEBlock(dBlockHash)
+			if err != nil {
+				return nil, NewInternalDatabaseError()
+			}
+			directoryBlockHeight = eBlock.GetDatabaseHeight()
+		} else {
+			return nil, NewObjectNotFoundError()
+		}
+	} else {
+		return nil, NewInvalidParamsError()
+	}
+
+	directoryBlockKeyMR, err := dbo.FetchDBKeyMRByHeight(directoryBlockHeight)
+	if err != nil || directoryBlockKeyMR == nil {
+		return nil, NewBlockNotFoundError()
+	}
+
+	response := new(AnchorsResponse)
+	response.Height = directoryBlockHeight
+	response.KeyMR = directoryBlockKeyMR.String()
+	response.Bitcoin = false
+	response.Ethereum = false
+
+	// Search for an AnchorRecord for the DBlock our entry is in
+	if dirBlockInfo, err := dbo.FetchDirBlockInfoByKeyMR(directoryBlockKeyMR); err != nil {
+		return nil, NewBlockNotFoundError()
+	} else if dirBlockInfo != nil {
+		// An anchor is already confirmed either Bitcoin or Ethereum for this directory block and was written back into Factom
+		dbi := dirBlockInfo.(*dbInfo.DirBlockInfo)
+
+		if dbi.BTCConfirmed {
+			bitcoin := new(BitcoinAnchorResponse)
+			bitcoin.TransactionHash = dbi.BTCTxHash.String()
+			bitcoin.BlockHash = dbi.BTCBlockHash.String()
+			response.Bitcoin = bitcoin
+		}
+		if dbi.EthereumConfirmed && !dbi.EthereumAnchorRecordEntryHash.IsSameAs(primitives.ZeroHash) {
+			anchorRecordEntry, err := dbo.FetchEntry(dbi.EthereumAnchorRecordEntryHash)
+			if err != nil {
+				return nil, NewCustomInternalError(err)
+			}
+			anchorRecordJSON := anchorRecordEntry.GetContent()
+			anchorRecord, err := anchor.UnmarshalAnchorRecord(anchorRecordJSON)
+			if err != nil {
+				return nil, NewCustomInternalError(err)
+			}
+			eth := new(EthereumAnchorResponse)
+			eth.DBHeightMax = int64(anchorRecord.DBHeightMax)
+			eth.DBHeightMin = int64(anchorRecord.DBHeightMin)
+			eth.WindowMR = anchorRecord.WindowMR
+			eth.RecordHeight = int64(anchorRecord.RecordHeight)
+			eth.ContractAddress = anchorRecord.Ethereum.ContractAddress
+			eth.TxID = anchorRecord.Ethereum.TxID
+			eth.BlockHash = anchorRecord.Ethereum.BlockHash
+			eth.TxIndex = anchorRecord.Ethereum.TxIndex
+
+			var allWindowKeyMRs []interfaces.IHash
+			for i := eth.DBHeightMin; i <= eth.DBHeightMax; i++ {
+				keyMR, err := dbo.FetchDBKeyMRByHeight(uint32(i))
+				if err != nil {
+					return nil, NewCustomInternalError(err)
+				}
+				allWindowKeyMRs = append(allWindowKeyMRs, keyMR)
+			}
+			eth.MerkleBranch = primitives.BuildMerkleBranchForHash(allWindowKeyMRs, directoryBlockKeyMR, true)
+			response.Ethereum = eth
```

Comment:
This looks like a code duplicate of what is below. Can we refactor into a function so we don't have to modify 2 pieces of the same code if we need a change?

Reply:
Thanks for the reminder there! Just updated it if ya wanted to take a look